feat/new product overview pages (#646)
hirosystems/docs: docs/clarinet/index.mdx

---
# The default id is the same as the one defined below. so not needed
title: Clarinet Overview
sidebar_label: Overview
custom_edit_url: null
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> <p class="p-3 rounded-lg bg-neutral-100 text-neutral-500 dark:bg-neutral-800 dark:text-neutral-400">Clarinet is a local development environment that offers everything you need to create and deploy smart contracts on Stacks.</p>

Reviewer: ```suggestion
<p class="p-3 rounded-lg bg-neutral-100 text-neutral-500 dark:bg-neutral-800 dark:text-neutral-400">Clarinet is a local development environment that offers everything you need to create, test, and deploy smart contracts on Stacks.</p>
```

<div class="mb-10"></div>

> With Clarinet, you can quickly setup your Clarity project through the command line tool. This covers every stage of the development cycle from generating new contracts, an interactive REPL and validating contracts to deployment on a devnet, testnet or mainnet.

Reviewer: ```suggestion
Using a simple command line interface, Clarinet enables you to build and iterate on your contracts locally before moving into production on mainnet. You can code with Clarity syntax support (the smart contract language for Stacks), run tests, debug your code line by line, and deploy your contracts to a local Stacks blockchain environment to rapidly iterate on your code.

Clarinet can be used locally or as a hosted service in the cloud via the [Hiro Platform](https://docs.hiro.so/platform).
```

    <p>Standardize your contract deployments with customizable plans that can execute multi-step processes automatically.</p>
  </a>
</div>

<div class="subSections my-8">
  <a href="/clarinet/guides/clarinet-integrate">
    <div class="subSectionTitle"><h3>Run a local development environment</h3><span>→</span></div>
    <p>Deploy your code in private blockchain environments to test and integrate your contracts.</p>
  </a>
  <a href="/clarinet/guides/analyze-with-check-checker">
    <div class="subSectionTitle"><h3>Analyzing contracts for vulnerabilities</h3><span>→</span></div>
    <p>Learn how to enable the `check_checker` feature to identify potentially dangerous areas in your contracts.</p>
  </a>
</div>

### Related Tools

**[Clarinet JS SDK](./guides/clarinet-js-sdk.md)**: Make testing in Clarinet even easier with the Clarinet SDK.

**[Hiro Platform](../platform/index.mdx)**: Skip the Clarinet installation and go straight to building in a hosted dev platform.

**[Stacks.js](../stacks.js/index.mdx)**: Don’t reinvent the wheel and build faster with a JS library that handles basic functions.

---

> Need help building with Clarinet? Reach out to us on the `#clarinet` channel on Discord under Hiro Developer Tools section. We also host a weekly office hours on Discord every Monday at 11am ET.

Reviewer: ```suggestion
Need help building with Clarinet? Reach out to us on the `#clarinet` channel on Discord under the [Hiro Developer Tools section](https://stacks.chat/). There’s also a [weekly office hours](https://events.stacks.co/event/HD16484710) on Discord every Thursday at 11am ET.
```
